docs/tutorials/reinforce_reacher_gym_v26.py

```diff
@@ -0,0 +1,249 @@
+   done, and additional info from the step. Due to reproducibility
+   issues that will be expanded on in a blog post soon, we have changed
+   the type definition to
+   tuple\ ``[ObsType, SupportsFloat, bool, bool, dict[str, Any]``]
+   adding an extra boolean value. This additional bool corresponds to
+   the older done, which has been changed to terminated and
+   truncated.These changes were introduced in Gym v26 (turned off by
+   default in v25).
+
+In this tutorial, we will be using the REINFORCE algorithm [1] to solve
+the task of *Reacher*. Reacher is a two-jointed robot arm. The goal is
+to move the robot’s end effector (called a fingertip) close to a target
+that is spawned at a random position.
+
+"""
+
+import gymnasium as gym
+import numpy as np
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.distributions import Categorical
+from torch.distributions.normal import Normal
+
+import pandas as pd
+import seaborn as sns
+import random
+
+import matplotlib.pyplot as plt
+from matplotlib.lines import Line2D
+plt.rcParams["figure.figsize"] = (10,5)
+
+import warnings
+warnings.filterwarnings("ignore")
+
+env = gym.make("Reacher-v4")
+
+class Policy_Network(nn.Module):
+
+    def __init__(self, observation_space, action_space) -> None:
+        super(Policy_Network, self).__init__()
+
+        """
+        Initializes a neural network that estimates the mean and standard deviation of a normal distribution from which an action is sampled from
+        
+        Args:
+        observation_space : int - Dimension of the observation space
+        action_space : int - Dimension of the action space
+        """
+
+        hidden_space1 = 16
+        hidden_space2 = 32
+
+        # Shared Network 
+        self.shared_net = nn.Sequential(nn.Linear(observation_space, hidden_space1),
+                                     nn.Tanh(),
+                                     nn.Linear(hidden_space1, hidden_space2),
+                                     nn.Tanh())
+        
+        # Policy Mean specific Linear Layer
+        self.policy_mean_net = nn.Sequential(nn.Linear(hidden_space2, action_space))
+
+        # Policy Std Dev specific Linear Layer
+        self.policy_stddev_net = nn.Sequential(nn.Linear(hidden_space2, action_space))
+
+    def forward(self, x):
+
+        """
+        Returns the mean and standard deviation of a normal distribution from which an action is sampled from
+        
+        Args:
+        x : tensor - Observation from the environment
+        """
+
+        x = x.float()
+
+        shared_features = self.shared_net(x)
+
+        action_means = self.policy_mean_net(shared_features)
+        
+        action_stddevs = self.policy_stddev_net(shared_features)
+        action_stddevs = torch.log(1 + torch.exp(action_stddevs)) # Kind a activation function log(1 + exp(x))
+
+        return action_means, action_stddevs
+
+class REINFORCE():
```

Comment:
Could you add a description of how the `REINFORCE` algorithm works, link to the original paper, explain the purpose of the hyperparameters